### PR TITLE
Reintroduce eslint for typescript

### PR DESCRIPTION
**Pull request by joshuaellis**
### What does it do?

* Adds eslint & prettier to docs
* Adds eslint to design-system package (which very closely mirrors the config for cloud)

### Why is it needed?

* Eslint is necessary for life.

### CODE REVIEW

File: packages/strapi-design-system/package.json [hunk 1]
```diff
@@ -61,6 +61,7 @@
     "build": "yarn build:prod",
     "build:prod": "vite build",
     "generate:types": "tsc --noEmit false --emitDeclarationOnly --declarationDir dist",
+    "lint": "eslint . --ext .js,.jsx,.tsx,.ts --fix",
```
Comment (gu-stav) [inline packages/strapi-design-system/package.json:64]: Should we be consistent with the CMS here and split `linting` from `fixing`, by setting up two tasks (e.g. for CI)?

```suggestion
    "lint": "eslint . --ext .js,.jsx,.tsx,.ts --fix",
    "format": "yarn lint -- --fix",
```

**Review by gu-stav — COMMENTED** [review]
Well done & thank you for closing items from my todo-list. 🪀

File: packages/strapi-design-system/package.json (hunk 1 above)
Comment (joshuaellis) [reply packages/strapi-design-system/package.json:64]: i only kept fix cause it was there, to be honest i don't mind omitting it and making the developer run the fix command lol. Your suggestion seems off so I can't commit it by the way.

Comment (gu-stav) [reply packages/strapi-design-system/package.json:64]: What do you mean by "off"?

Comment (joshuaellis) [reply packages/strapi-design-system/package.json:64]: You didn't remove `--fix` 👀  sneaky

Comment (gu-stav) [reply packages/strapi-design-system/package.json]: Yes, I'm guilty of editing the message afterward. Sorry 🙈

**Review by gu-stav — APPROVED** [review]

**Pull request merged**